### PR TITLE
adds febs configuration file capability

### PR DESCRIPTION
**Pull request by DebugXYZ**
Work in progress, have a better way to unit test using the virtual filesystem instead of adding febs-config to febs repository

Also working out a better way to write febsConfMerge without the eslint disable

This is for issue #27

### CODE REVIEW

**Comment by DebugXYZ** [comment]
Unit test failures look like they may have to do with how travis is setup, but will verify

File: index.js [hunk 1]
```diff
@@ -46,6 +47,54 @@ module.exports = function init(conf = {}) {
...
+    return R.merge(febsConfig, febsConfigFileJSON);
+  };
+
+  const getPackageName = () => {
+    const projectPackageJson = path.join(projectPath, 'package.json');
+    return require(projectPackageJson).name;
+  };
+
+  // Applies febs-config to the webpack configuration
+  const febsConfigMerge = (febsConfig, wpConf) => {
```
[the remark this answers is not in the document]
Comment (peripateticus) [reply index.js]: You can ignore this one.

File: index.js [hunk 2]
```diff
@@ -46,6 +47,50 @@ module.exports = function init(conf = {}) {
...
+
+  const getFebsConfig = () => {
+    const febsConfig = getFebsConfigDefaults();
+
+    const febsConfigPath = path.resolve(projectPath, './febs-config.json');
+
+    let febsConfigFileJSON;
+    if (fs.existsSync(febsConfigPath)) {
+      febsConfigFileJSON = require(febsConfigPath);
+    } else if (febsConfigArg && (febsConfigArg.output || febsConfigArg.entry)) {
```
Comment (DebugXYZ) [inline index.js]: https://github.com/rei/febs/issues/59

File: index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DebugXYZ) [reply index.js]: Great call out, as discussed internally this will be fixed in another PR https://github.com/rei/febs/issues/60

File: ReadMe.md [hunk 3]
```diff
@@ -182,7 +182,7 @@ You can override or create new configurations through webpack directly where nec
 think others might need the override please file a ticket or reach out for [support](#support).
 Where you can, attempt to avoid using this feature to reduce duplication of work.
 
-To customize your build simply by creating a `webpack.overrides.conf.js` at the root of your npm
+To customize your build, creating a `webpack.overrides.conf.js` at the root of your npm
```
Comment (peripateticus) [inline ReadMe.md]: creating -> create

**Pull request merged**